### DISCUSSION TITLE
Host Mailu on a subdomain : how to configure hostnames ?

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [Mailu/Mailu#3121] [User support]
Hi,

I can't host mailu web ui (admin and roundcube) on my root domain because I already have another service here. So I host it on `mail.mydomain.com`, using caddy reverse proxy. I am using `mail-letsencrypt` TLS flavor, because caddy already handles TLS for the web ui but I still need mailu to automatically get a cert for emails.

But I still want my email domain to be `mydomain.com`, so that my addresses look like `admin@mydomain.com` and not `admin@mail.mydomain.com`. So I need to set the variable `DOMAIN` to `DOMAIN=mydomain.com`.

I then have 2 options and none seem to be working.

Option 1: If I set `HOSTNAMES=mydomain.com`, then letsencrypt fails at retrieving a cert for the emails, because let's encrypt expects the acme challenge to be exposed on `mydomain.com`, where it is actually on `mail.mydomain.com` because of the reverse proxy for web ui.

Option 2: If I set `HOSTNAMES=mail.mydomain.com`, then letsencrypt succeeds at retrieving a cert, but I can't receive any email. In that option, the DNS domain entries mailu admin ui tells me to set are all about `mail.mydomain.com` (against `mydomain.com` in the 1st option). I suspect this is the reason why I can't receive any email, because mailu listens on port 25 but only on domain `mail.mydomain.com` and not on `mydomain.com`, on which emails are sent at.

I have the following questions:
1. What would be my way out of this ?
2. Am I wrong about something about options 1 and 2 ?
3. Is there a way out option 1 or 2 I didn't see ?
4. Is there another option ?
5. Is it possible at all to host mailu while having the webui on a subdomain ?

Thanks in advance for any answer, have a great day

**Comment by ghostwheel42** [comment]
You can set DOMAIN and HOSTAMES to mail.yourdomain.com
This does not prevent you from using yourdomain.com as your email domain.
Just set the correct MX, SPF and other dns records and you are ready to go. 

Alex

**Reply by LeVraiRoiDHyrule** [reply]
Option 2 was the way to go. The reason why I couldn't receive emails is because the port opened on docker was written "127.0.0.1:25:25". I'm not sure why it wasn't working, but replacing by juste 25:25 fixed it.
